pkg/controllers/provisioner/v1alpha1/allocation/greedyallocator.go

```diff
@@ -45,25 +49,72 @@ func (a *GreedyAllocator) Allocate(pods []*v1.Pod) error {
 
 type SchedulingGroup struct {
 	Pods        []*v1.Pod
-	Constraints cloudprovider.CapacityConstraints
+	Constraints *cloudprovider.CapacityConstraints
 }
 
-func (a *GreedyAllocator) getSchedulingGroups(pods []*v1.Pod) []SchedulingGroup {
-	groups := []SchedulingGroup{}
-
+func (a *GreedyAllocator) getSchedulingGroups(pods []*v1.Pod) []*SchedulingGroup {
+	groups := []*SchedulingGroup{}
 	for _, pod := range pods {
+		added := false
 		for _, group := range groups {
-			if a.matchesGroup(pod, group) {
+			if a.matchesGroup(group, pod) {
+				addPodResourcesToList(group.Constraints.Resources, pod)
 				group.Pods = append(group.Pods, pod)
+				added = true
 				break
 			}
 		}
+		if added {
+			continue
+		}
+		groups = append(groups, schedulingGroupForPod(pod))
 	}
-
 	return groups
 }
 
 // TODO
-func (a *GreedyAllocator) matchesGroup(pod *v1.Pod, group SchedulingGroup) bool {
-	return true
+func (a *GreedyAllocator) matchesGroup(group *SchedulingGroup, pod *v1.Pod) bool {
+	return false
+}
+
+func schedulingGroupForPod(pod *v1.Pod) *SchedulingGroup {
+	group := &SchedulingGroup{
+		Constraints: &cloudprovider.CapacityConstraints{
+			Resources:    calculateResourcesForPod(pod),
+			Overhead:     calculateOverheadResources(),
+			Architecture: getSystemArchitecture(pod),
+			Zone:         getAvalabiltyZoneForPod(pod),
+		},
+	}
+	return group
+}
+
+func calculateResourcesForPod(pod *v1.Pod) v1.ResourceList {
+	resourceList := v1.ResourceList{
+		v1.ResourceCPU:    resource.MustParse("0"),
```

Comment:
IIUC, if you call container.resources.CPU() and it isn't set, it's equivalent to 0. You should just be able to initialize it directly from the pod.